sycl/include/sycl/ext/intel/experimental/fpga_mem/fpga_mem.hpp

```diff
@@ -0,0 +1,138 @@
+//==----------- fpga_mem.hpp - SYCL fpga_mem extension -----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/access/access.hpp> // for address_space
+#include <sycl/exception.hpp>     // for make_error_code
+#include <sycl/ext/intel/experimental/fpga_mem/properties.hpp> // for num_banks
+#include <sycl/ext/oneapi/properties/properties.hpp> // for properties_t
+// #include <sycl/pointers.hpp>                         // for decorated_gl...
+
+#include <cstddef>     // for ptrdiff_t
+#include <type_traits> // for enable_if_t
+#include <utility>     // for declval
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::intel::experimental {
+// shorthand used with properties
+namespace oneapi_exp = ext::oneapi::experimental;
```

Comment:
Please move this into `detail` if we need it. We shouldn't leak these into user-space.